src/server/connection.cpp

```diff
@@ -140,10 +145,15 @@ std::vector<char> Connection::compress_buffers(const std::vector<char> &uncompre
     boost::iostreams::filtering_ostream gzip_stream;
     gzip_stream.push(boost::iostreams::gzip_compressor(compression_parameters));
     gzip_stream.push(boost::iostreams::back_inserter(compressed_data));
-    gzip_stream.write(&uncompressed_data[0], uncompressed_data.size());
+    const auto &result = gzip_stream.write(&uncompressed_data[0], uncompressed_data.size());
+    if (!result)
+    {
+        throw util::RuntimeError(
```

Comment:
Hmm, while I think it's good to check for error here, I'm not sure that throwing is the right solution - this will kill the server process, as I don't think we have a high-level `catch` on a per-request basis.

Write errors like this one should cause the client on the other end of the connection to receive an error message, but probably shouldn't cause the `osrm-routed` process to terminate.

Reply:
> Hmm, while I think it's good to check for error here, I'm not sure that throwing is the right solution - this will kill the server process, as I don't think we have a high-level `catch` on a per-request basis.
> 

Argh, of course! Sorry about that.

> Write errors like this one should cause the client on the other end of the connection to receive an error message, but probably shouldn't cause the `osrm-routed` process to terminate.

I'm probably missing something, but should the error message really trickle down all the way to the client? If I understand it correctly, `compress_buffers()` is only trying to apply the compression method set in the HTTP Accept-Encoding header, which, if I understand [this description](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Encoding) correctly, is more of a suggestion– "if you choose to compress, I understand the following methods". If the server fails to compress, no harm done to the client.

That being said, the compression failure should probably still be logged at the server?

Reply:
Sorry for the belated answer, cleaning up the old PRs. :sparkles: 

@Lotterleben My reading of that description is the same, though I think we would need to return plaintext if compression fails, right?

So if anyone feels like picking this up again:
- We need to probably surface to the caller of this function that compression failed (either through the return value or through a custom exception).
- The call code needs to handle that and return plaintext instead.